### PR TITLE
Bug 1939855: controllers/updateservice_controller.go: handle init container changes

### PR DESCRIPTION
**Pull request by jottofar**
such as graphDataImage updates.

### CODE REVIEW

File: controllers/updateservice_controller.go [hunk 1]
```diff
@@ -389,6 +391,20 @@ func (r *UpdateServiceReconciler) ensureDeployment(ctx context.Context, reqLogge
...
+	initContainers := updated.Spec.Template.Spec.InitContainers
+	for i := range initContainers {
+		var original *corev1.Container
+		switch initContainers[i].Name {
+		case "graph-data":
+			original = resources.graphDataInitContainer
+		default:
+			reqLogger.Info("encountered unexpected init container in pod", "Container.Name", initContainers[i].Name)
+			continue
+		}
```
Comment (petr-muller) [inline controllers/updateservice_controller.go:403]: Looks more wordy than I'd expect, why not simply:

```go
if initContainers[i].Name != `graph-data` {
  logger.Info
  continue
}
original = ...
initContainers[i].Image = original.Image
...
```

Is it likely that we grow more `case`s?

Comment (jottofar) [reply controllers/updateservice_controller.go:403]: I agree. I just cut-and-pasted it and been concentrating on trying to get it to test out but I'll change it.

File: controllers/updateservice_controller.go [hunk 2]
```diff
@@ -389,6 +391,19 @@ func (r *UpdateServiceReconciler) ensureDeployment(ctx context.Context, reqLogge
 		containers[i].LivenessProbe = original.LivenessProbe
 		containers[i].ReadinessProbe = original.ReadinessProbe
 	}
+	initContainers := updated.Spec.Template.Spec.InitContainers
+	for i := range initContainers {
+		var original *corev1.Container
+		if initContainers[i].Name == "graph-data" {
+			original = resources.graphDataInitContainer
+		} else {
+			reqLogger.Info("encountered unexpected init container in pod", "Container.Name", initContainers[i].Name)
```
Comment (wking) [inline controllers/updateservice_controller.go:400]: why are we keeping these?  Can we stomp them off?

Comment (wking) [reply controllers/updateservice_controller.go:400]: like:

```go
updated.Spec.Template.Spec.Initcontainers = []whateverType{resources.graphDataInitContainer}
```

or some such?

Comment (wking) [reply controllers/updateservice_controller.go:400]: The Cincinnati Deployment is our operand.  We have opinions about its init containers.  Other folks might have opinions too, and attempt to inject additional init containers.  But trying to share ownership of a single Deployment is sticky, and in the CVO we've had many, many bugs as a result of initially attempting a sharing approach (openshift/cluster-version-operator#830, openshift/cluster-version-operator#804, openshift/cluster-version-operator#674, openshift/cluster-version-operator#654, openshift/cluster-version-operator#650, openshift/cluster-version-operator#563, openshift/cluster-version-operator#549 , ...), and no reports that I can remember of any users benefiting from the sharing.

**Review by PratikMahajan — COMMENTED** [review]
/retest

**Comment by PratikMahajan** [comment]
/override ci/prow/operator-e2e

operator-e2e is pinned to a specific commit which does not contain the fix for sha-sum mismatch. 
ci/prow/operator-e2e-latest-osus  uses the latest commit from cincinnati and has passed the tests.

**Review by wking — APPROVED** [review]
/lgtm

**Comment by jottofar** [comment]
/override ci/prow/operator-e2e

operator-e2e is pinned to a specific commit which does not contain the fix for sha-sum mismatch.
ci/prow/operator-e2e-latest-osus uses the latest commit from cincinnati and has passed the tests.

**Pull request merged**